Build fail real:
```
======================================================================
FAIL: test_case_tile_suite (corehq.apps.app_manager.tests.test_suite.SuiteTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/test_suite.py", line 341, in test_case_tile_suite
    self._test_generic_suite("app_case_tiles", "suite-case-tiles")
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/util.py", line 71, in _test_generic_suite
    self.assertXmlEqual(self.get_xml(suite_tag), app.create_suite())
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/util.py", line 32, in assertXmlEqual
    _check_shared(expected, actual, LXMLOutputChecker(), "xml")
  File "/home/travis/build/dimagi/commcare-hq/corehq/apps/app_manager/tests/util.py", line 127, in _check_shared
    raise AssertionError(message)
AssertionError: XML mismatch
--- want.xml
+++ got.xml
@@ -290,7 +290,7 @@
     </command>
     <instance id="casedb" src="jr://instance/casedb"/>
     <session>
-      <datum detail-inline="m0_case_long" detail-persistent="m0_case_short" detail-select="m0_case_short" id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='child'][@status='open']" value="./@case_id"/>
+      <datum detail-confirm="m0_case_long" detail-persistent="m0_case_short" detail-select="m0_case_short" id="case_id" nodeset="instance('casedb')/casedb/case[@case_type='child'][@status='open']" value="./@case_id"/>
     </session>
   </entry>
   <entry>
```

--- a/corehq/apps/app_manager/suite_xml/sections/entries.py
+++ b/corehq/apps/app_manager/suite_xml/sections/entries.py
@@ -822,6 +822,7 @@ class EntriesHelper(object):
         for detail_type, detail, enabled in module.get_details():
             if (detail.persist_tile_on_forms and (detail.use_case_tiles or detail.custom_xml) and enabled):
                 detail_persistent = id_strings.detail(detail_module, detail_type)
-                detail_inline = self.details_helper.get_detail_id_safe(detail_module, 'case_long')
+                if detail.pull_down_tile:
+                    detail_inline = self.details_helper.get_detail_id_safe(detail_module, 'case_long')
                 break
         return detail_persistent, detail_inline

--- a/corehq/apps/app_manager/tests/test_suite.py
+++ b/corehq/apps/app_manager/tests/test_suite.py
@@ -401,6 +401,71 @@ class SuiteTest(SimpleTestCase, TestXmlMixin, SuiteMixin):
             "./entry/session"
         )
 
+    def test_persistent_case_tiles_in_advanced_forms(self):
+        """
+        Test that the detail-persistent attributes is set correctly when persistent
+        case tiles are used on advanced forms.
+        """
+        factory = AppFactory()
+        module, form = factory.new_advanced_module("my_module", "person")
+        factory.form_requires_case(form, "person")
+        module.case_details.short.custom_xml = '<detail id="m0_case_short"></detail>'
+        module.case_details.short.persist_tile_on_forms = True
+        suite = factory.app.create_suite()
+
+        # The relevant check is really that detail-persistent="m0_case_short"
+        # but assertXmlPartialEqual xpath implementation doesn't currently
+        # support selecting attributes
+        self.assertXmlPartialEqual(
+            """
+            <partial>
+                <datum
+                    detail-confirm="m0_case_long"
+                    detail-persistent="m0_case_short"
+                    detail-select="m0_case_short"
+                    id="case_id_load_person_0"
+                    nodeset="instance('casedb')/casedb/case[@case_type='person'][@status='open']"
+                    value="./@case_id"
+                />
+            </partial>
+            """,
+            suite,
+            "entry/session/datum"
+        )
+
+    def test_persistent_case_tiles_in_advanced_module_case_lists(self):
+        """
+        Test that the detail-persistent attributes is set correctly when persistent
+        case tiles are used on advanced module case lists
+        """
+        factory = AppFactory()
+        module, form = factory.new_advanced_module("my_module", "person")
+        factory.form_requires_case(form, "person")
+        module.case_list.show = True
+        module.case_details.short.custom_xml = '<detail id="m0_case_short"></detail>'
+        module.case_details.short.persist_tile_on_forms = True
+        suite = factory.app.create_suite()
+
+        # The relevant check is really that detail-persistent="m0_case_short"
+        # but assertXmlPartialEqual xpath implementation doesn't currently
+        # support selecting attributes
+        self.assertXmlPartialEqual(
+            """
+            <partial>
+                <datum
+                    detail-confirm="m0_case_long"
+                    detail-persistent="m0_case_short"
+                    detail-select="m0_case_short"
+                    id="case_id_case_person"
+                    nodeset="instance('casedb')/casedb/case[@case_type='person'][@status='open']"
+                    value="./@case_id"
+                />
+            </partial>
+            """,
+            suite,
+            'entry/command[@id="m0-case-list"]/../session/datum',
+        )
+
     def test_subcase_repeat_mixed(self):
         app = Application.new_app(None, "Untitled Application", application_version=APP_V2)
         module_0 = app.add_module(Module.new_module('parent', None))

